yb-voyager/src/queryparser/traversal_plpgsql.go

```diff
@@ -0,0 +1,171 @@
+	PERFORM pg_advisory_lock(receiver_id);
+
+	-- Deduct the amount from the sender's account
+	UPDATE accounts
+	SET balance = balance - transfer_amount
+	WHERE account_id = sender_id;
+
+	-- Add the amount to the receiver's account
+	UPDATE accounts
+	SET balance = balance + transfer_amount
+	WHERE account_id = receiver_id;
+
+	-- Release the advisory locks
+	PERFORM pg_advisory_unlock(sender_id);
+	PERFORM pg_advisory_unlock(receiver_id);
+
+END;
+$$;
+
+parsed json -
+	{
+	  "PLpgSQL_function": {
+	    "datums": [
+	      ...
+	    ],
+	    "action": {
+	      "PLpgSQL_stmt_block": {
+	        "lineno": 2,
+	        "body": [
+	          {
+	            "PLpgSQL_stmt_perform": {
+	              "lineno": 4,
+	              "expr": {
+	                "PLpgSQL_expr": {
+	                  "query": "SELECT pg_advisory_lock(sender_id)",
+	                  "parseMode": 0
+	                }
+	              }
+	            }
+	          },
+	          {
+				... similar to above
+	          },
+	          {
+	            "PLpgSQL_stmt_execsql": {
+	              "lineno": 8,
+	              "sqlstmt": {
+	                "PLpgSQL_expr": {
+	                  "query": "UPDATE accounts \n    SET balance = balance - transfer_amount \n    WHERE account_id = sender_id",
+	                  "parseMode": 0
+	                }
+	              }
+	            }
+	          },
+	          {
+				.... similar to above
+	          },
+	          {
+	            ... similar to below
+	          },
+	          {
+	            "PLpgSQL_stmt_perform": {
+	              "lineno": 19,
+	              "expr": {
+	                "PLpgSQL_expr": {
+	                  "query": "SELECT pg_advisory_unlock(receiver_id)",
+	                  "parseMode": 0
+	                }
+	              }
+	            }
+	          },
+	          {
+	            "PLpgSQL_stmt_return": {}
+	          }
+	        ]
+	      }
+	    }
+	  }
+	}
+*/
+func TraversePlPgSQLActions(action interface{}, plPgSqlStatements *[]string) {
+	actionMap, ok := action.(map[string]interface{})
+	if !ok {
+		//In case the value of a field is not a <key , val> but a list of <key, val> e.g. "body"
+		lists, ok := action.([]interface{})
+		if ok {
+			for _, l := range lists {
+				TraversePlPgSQLActions(l, plPgSqlStatements)
+			}
+		}
+		return
+	}
+
+	for k, v := range actionMap {
+		switch k {
+		// base case of recursive calls to reach this PLPGSQL_EXPR field in json which will have "query" field with statement
+		case PLPGSQL_EXPR: 
+			expr, ok := v.(map[string]interface{})
+			if ok {
+				query, ok := expr[QUERY]
+				if ok {
+					q := formatExprQuery(query.(string)) // formating the query of parsed json if required
+
+					*plPgSqlStatements = append(*plPgSqlStatements, q)
+				}
+			}
+		default:
+			TraversePlPgSQLActions(v, plPgSqlStatements)
+		}
+	}
+}
+
+// Function to format the PLPGSQL EXPR query from the json string
+func formatExprQuery(q string) string {
```

Comment:
If I remember, parser has a format option also, right? The reason we don't do that here is because of the limitations, parsing itself can fail, correct?

Reply:
I am not sure about the format option from parser, I don't think there is any but will check. Anyways yes we can't use that here as you mentioned because of limitations.